api/src/api/language/vocab.js

```diff
@@ -103,4 +103,56 @@ router.post(
   }),
 );
 
+/**
+ * Updates the fields for multiple vocab items
+ */
+router.put(
+  '/',
+  requireAuthentication,
+  requireLanguageAuthorization,
+  errorWrap(async (req, res) => {
+    const { lesson_id, vocab_updates } = req.body;
+    if (!lesson_id || !vocab_updates) {
+      return sendResponse(res, 400, ERR_MISSING_OR_INVALID_DATA);
+    }
+
+    // Checks if the ids are valid
+    const isValid = await checkIds({ lesson_id });
+
+    if (!isValid) {
+      return sendResponse(res, 400, ERR_MISSING_OR_INVALID_DATA);
+    }
+
+    /* Get the lesson data from MongoDB */
+    let lesson = await models.Lesson.findById(lesson_id);
+
+    for (let i = 0; i < vocab_updates.length; i++) {
+      /* Obtain the index of the vocab item that we want to update */
+      const vocabIndex = getVocabIndexByID(vocab_updates[i]._id, lesson);
+
+      if (vocabIndex === NOT_FOUND_INDEX) {
+        return sendResponse(res, 404, 'Vocab item not found');
+      }
+
+      /* Using the index, apply changes to the lesson data */
+      let vocabData = lesson.vocab[vocabIndex];
+
+      for (var key in vocab_updates[i]) {
```

Comment:
I think there exists `patchDocument` helper method with the same functionality as this piece of code. The method is in `api/src/utils/languageHelper.`

Reply:
I changed it to use `patchDocument` instead